package amber.collision_avoidance_proto;
option java_package = "pl.edu.agh.amber.collisionavoidance.proto";
option java_outer_classname = "CollisionAvoidanceProto";

import "drivermsg.proto";

extend amber.DriverMsg {
    optional bool setSpeed = 90;
    optional bool getSpeed = 91;
    optional bool getSpeedAndScan = 92;
    optional bool getScan = 93;
    optional MotorsSpeed motorsSpeed = 94;
    optional Scan scan = 95;
}

message MotorsSpeed {
    optional int32 frontLeftSpeed = 1;
    optional int32 frontRightSpeed = 2;
    optional int32 rearLeftSpeed = 3;
    optional int32 rearRightSpeed = 4;
}

message Scan {
    repeated double angles = 1 [packed = true];
    repeated int32 distances = 2 [packed = true];
}
